src/models/network/NetworkBaseController.js

```diff
@@ -111,48 +111,57 @@ export default class NetworkBaseController {
   }
 
   checkLocalContractsDeployed(throwIfFail = false) {
-    let msg;
+    this._handleErrorMessage(this._errorForLocalContractsDeployed(), throwIfFail);
```

Comment:
IMO checking and handling are kind of disruptive here, it is kind of weird to call a handler as a first step. I would have expected a condition at least.

Reply:
Agree, changed it